src/celeritas/user/RootStepWriter.cc

```diff
@@ -139,6 +139,10 @@ void RootStepWriter::process_steps(HostStepState state)
     } while (0)
 
     CELER_EXPECT(state.steps);
+    if (state.stream_id.get() != 0)
```

Comment:
Type-safe comparison:
```suggestion
    if (state.stream_id != StreamId{0})
```